### PR TITLE
Implement Vendors Option

### PR DESCRIPTION
**Pull request by switters**
### Problem

Retsly OAuth now accepts ```vendors``` argument. Retsly-js-auth should accept ```vendors``` as an initiating argument.

### Solution

- [x] Extend the url created by retsly-js-auth, adding ```vendors``` to the url parameters if included in initialization arguments. 4085b3d

##### Auxiliary Changes

- ```scopes``` -> ```scope``` for consistency with OAuth parameters ef9bb39
- Added ```node_modules``` to ```.gitignore``` b9e6f55

##### Bug Fixes

- Pinned ```mocha-phantomjs``` version at 3.6.0 to prevent test errors that occurred with 4.0.0 of that module. f2a1022
 
### Deployment

- [x] Update docs to detail ```vendors``` option 885b118
- [x] Bump version to 0.7.3 eace134
- [ ] Create new release for 0.7.3
- [ ] www ref'd version of retsly-js-auth will be bumped in Retsly/www#478

Ref: Retsly/www#478

### CODE REVIEW

**Comment by switters** [comment]
@Jkhwan @CMClay Review?

**Comment by Jkhwan** [comment]
:+1:

**Comment by CMClay** [comment]
:+1:

**Comment by switters** [comment]
:rocket:

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,3 @@
 components
 build
+node_modules
```
Comment (CMClay) [inline .gitignore]: Wow... how was this not here?

Comment (switters) [inline .gitignore]: lol right?

**Pull request merged**